### ISSUE TITLE
Root listen path 302 redirect

### ISSUE THREAD

**Issue by drpebcak** [TykTechnologies/tyk#1538]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
After upgrading from tyk gateway 1.3.4 to 1.5.3, my existing apis 302 requests from the root of the listen path to the path specified in my target url:

listen path = '/example_api/'
target url = 'http://api.example.com/api/'
```
curl http://example.com/example_api/ -I
HTTP/1.1 302 Found
Content-Type: text/plain; charset=utf-8
Date: Wed, 14 Mar 2018 17:29:44 GMT
Location: /api/
X-Ratelimit-Limit: -1
X-Ratelimit-Remaining: -1
X-Ratelimit-Reset: 1507675133
Connection: keep-alive
```

**What is the expected behavior?**
I should be able to do a GET on the root of my listen path and be returned whatever my app is serving there.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**
In my experience, any API I have set up is no longer accessible through tyk at the root of my listen path.

**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
It used to work as expected in 1.3.4, but after upgrading to 1.5.3 I see this issue

**Comment by buger** [comment]
Thank you for reporting! 

In your example it redirects to "/api/" urls, but I assume it just a mock for this github issue. Where it actually redirects, what is the pattern? What if you try without trailing "/" like `curl http://example.com/example_api`?

**Comment by drpebcak** [comment]
Hey,

That's actually the path for my target url (only the domain was changed).

If I curl without the trailing slash it 404s, because my listen path in tyk is configured with the trailing slash. If I change my config so the listen path is `/example_api` without that slash, visiting `http://example.com/example_api`  302s to `/api/` still.

**Comment by buger** [comment]
So, it looks like Tyk ignores trailing URL of your target URL, and your app trigger redirect, since it configured to work with trailing slash. Correct?

**Comment by drpebcak** [comment]
@buger yeah it does look like that. This behavior is new since the upgrade I did to tyk. Is there a workaround or something I can do to get around this?

**Comment by buger** [comment]
I would suggest setting the following option:
```"http_server_options": {
      "override_defaults": true
      "skip_url_cleaning": true
}```

**Comment by drpebcak** [comment]
Hey, thanks for the suggestion. I just set those options in my gateway config, but the same thing is happening.

**Comment by drpebcak** [comment]
@buger any more ideas on this?

**Comment by asoorm** [comment]
If you could send an API definition or a gist of an API definition, redacting sensitive info, it may help us to debug the issue.

**Comment by drpebcak** [comment]
@asoorm  here you go: https://gist.github.com/drpebcak/6bcf113ac02205818d6f0f569a0f3f64

**Comment by asoorm** [comment]
This is strange - I've imported your api definition, replacing a few nulls with empty array or objects where appropriate. Then I've changed auth method to Auth Token for speed - and setup the upstream to be http://httpbin.org/get

https://gist.github.com/asoorm/859f3a95b6b89de3b318c8bc79a5cf34

When I create a key for the API and send a request to cloud gateway - it proxies the request without issue.

```
curl --request GET \
  --url https://GATEWAY_ADDRESS/example_api/ \
  --header 'Authorization: AUTH_TOKEN'
```

I get the following response from httpbin

```
< HTTP/1.1 200 OK
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Origin: *
< Content-Type: application/json
< Date: Tue, 20 Mar 2018 20:57:46 GMT
< Server: meinheld/0.6.1
< Via: 1.1 vegur
< X-Powered-By: Flask
< X-Processed-Time: 0
< X-Ratelimit-Limit: -1
< X-Ratelimit-Remaining: -1
< X-Ratelimit-Reset: 1521577590
< Content-Length: 511
< Connection: keep-alive
<
{
  "args": {},
  "headers": {
    "Accept": "*/*",
    "Accept-Encoding": "gzip",
    "Authorization": "AUTH_TOKEN",
    "Connection": "close",
    "Host": "httpbin.org",
    "User-Agent": "curl/7.54.0",
    "X-Scheme": "https"
  },
  "origin": "IP1, IP2, IP3...",
  "url": "http://httpbin.org/get"
}
```

**Comment by drpebcak** [comment]
I think the main difference here is that when hitting my api, directly, it will redirect `http://example-api.example-api.rancher.internal:8080/api` to `http://example-api.example-api.rancher.internal:8080/api/` (note the trailing slash). It looks like tyk is not sending requests with the trailing slash like it used to, even though it's in my `target_url`.

It works ok for httpbin because they don't redirect to the trailing slash (in their case, it actually doesn't work at all with a trailing slash).

**Comment by drpebcak** [comment]
@buger any more ideas or fixes for this one?

**Comment by buger** [comment]
Added new `proxy.disable_strip_prefix` boolean variable

**Comment by buger** [comment]
https://github.com/TykTechnologies/tyk/pull/1972 was merged to `release-2.7` branch
<details><summary></summary>Created via API</details>

**Comment by buger** [comment]
https://github.com/TykTechnologies/tyk/pull/1972 was merged to `release-2.8` branch
<details><summary></summary>Created via API</details>

**Comment by drpebcak** [comment]
Thanks!
